android/hybridcommon/src/main/java/com/revenuecat/purchases/hybridcommon/common.kt

```diff
@@ -574,6 +578,17 @@ fun getPromotionalOffer(): ErrorContainer {
 
 // region private functions
 
+private fun String.toPurchasesAreCompletedBy(): PurchasesAreCompletedBy? {
+    return when (this) {
+        "REVENUECAT" -> PurchasesAreCompletedBy.REVENUECAT
+        "MY_APP" -> PurchasesAreCompletedBy.MY_APP
+        else -> {
+            Log.e("PurchasesHybridCommon", "Unrecognized PurchasesAreCompletedBy: $this")
+            null
+        }
+    }
+}
```

Comment:
We can use `enumValueOf()` or `PurchasesAreCompletedBy.valueOf()` to avoid having to write out the literal enum entries. 

```kotlin
private fun String.toPurchasesAreCompletedBy(): PurchasesAreCompletedBy? {
    return try {
        enumValueOf<PurchasesAreCompletedBy>(this)
    } catch (e: IllegalArgumentException) {
        null
    }
}
```
I'd also be okay not catching the exception and failing fast. It means we have made a mistake integrating PHC in a hybrid, and a crash is easier to notice (and thus more likely to get fixed) than unusual behavior due to a null `PurchasesAreCompletedBy`.

Reply:
Good point! Will do that change.

As for crash vs log, we've been very reticent to do crashes even for things that are likely our mistake like this... Happy to discuss this policy more though!

Reply:
Done in https://github.com/RevenueCat/purchases-hybrid-common/pull/859/commits/dab8a3fce431c80955f4f707c3b5bc5bbf90a9f0

Reply:
Yea let's not deviate from a previously discussed policy here! I would like to learn a bit more about it, but let's do that in our weekly for instance.